tokenizers/src/pre_tokenizers/whitespace.rs

```diff
@@ -31,9 +34,50 @@ impl PreTokenizer for Whitespace {
     }
 }
 
-#[derive(Serialize, Deserialize)]
+// manually implement serialize / deserialize because Whitespace is not a unit-struct but is
+// serialized like one.
+impl Serialize for Whitespace {
```

Comment:
The macro can't be used because `Whitespace` is not a valid expression to construct that struct, but at least the `Serialize` impl is now gone with your other suggestion!